src/utils/SpokeUtils.ts

```diff
@@ -102,3 +103,66 @@ export function isZeroAddress(address: string): boolean {
 export function getMessageHash(message: string): string {
   return isMessageEmpty(message) ? ZERO_BYTES : keccak256(message as Hex);
 }
+
+export async function findInvalidFills(spokePoolClients: {
+  [chainId: number]: SpokePoolClient;
+}): Promise<InvalidFill[]> {
+  const invalidFills: InvalidFill[] = [];
+
+  // Iterate through each spoke pool client
+  for (const spokePoolClient of Object.values(spokePoolClients)) {
+    // Get all fills for this client
+    const fills = spokePoolClient.getFills();
+
+    // Process each fill
+    for (const fill of fills) {
+      // Skip fills with unsafe deposit IDs
+      if (isUnsafeDepositId(fill.depositId)) {
+        continue;
+      }
```

Comment:
We can add a todo here to circle back to it as a follow-up, because we do ultimately want to detect and log about these (but only after some time period, to account for pre-fills).